src/macports1.0/macports.tcl

```diff
@@ -699,12 +699,16 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
     # set up platform info variables
     set os_arch $tcl_platform(machine)
     # Set os_arch to match `uname -p`
-    if {$os_arch eq "Power Macintosh"} {
-        set os_arch "powerpc"
-    } elseif {$os_arch eq "i586" || $os_arch eq "i686" || $os_arch eq "x86_64"} {
-        set os_arch "i386"
-    } elseif {$os_arch eq "arm64"} {
-        set os_arch "arm"
+    switch -regexp -- $os_arch {
```

Comment:
There's really no reason to involve regexes here, `-glob` and some fallthrough cases would be preferable.

Reply:
Thanks @jmroot. Done.